### DISCUSSION TITLE
Cypress Framework , a product disappears from the bucket

### DISCUSSION THREAD

**Discussion by vadim3000** [cypress-io/cypress#22792] [Questions and Help]
Hello there, I'm Vadim QA Engineer

I'm currently trying to automate my job website, https://meinfoto.de/   . The Back-End writen in Java.
The problem is. When the user is on the shopping cart page and clicks on go to the next page, the product is despairing from the bucket,

After clarification with Back-End dev, the suggestion was, might the problem with cookies. JSESSION cookie gets overwritten by Cypress framework.
Also I  added to blockHosts:     "*google-analytics.com",  "*googletagmanager.com",     "*google.com/ads/",

Video:

https://user-images.githubusercontent.com/72452790/178998446-7a1848f9-3182-48f7-80b2-061c6630b72a.mp4

I haven't had this type of problem with Java and Selenium-based framework. 

Thanks

**Comment by jjhelguero** [comment]
It looks like your tests depend on the previous one. If that is the case, that is risky and not good practice. Cypress will clear cache and cookies between each test, so I imagine that is the reason the product disappears from the basket. Can you share your test code?

**Comment by vadim3000** [comment]
Hello, yes

 /// <reference types="cypress"/>
/// <reference types=”cypress-xpath” />

describe('Test on mf', () =>  {

  it('the user visiting Canvas LP', () => {
      cy.visit('https://www.meinfoto.de/foto-in-gross/foto-auf-leinwand.jsf')
      cy.wait(500)
      cy.xpath("//span[contains(text(),'Alle akzeptieren')]")
      .click()
  })

  it('The click on CTA button and uploads the file', () => {

      cy.wait(500)
      cy.xpath('//div[@class="upload-button"]//input[@type="file"]')
      .attachFile("test_image_1.jpg")
  })

  it('the user on the configurator page and clicks on "Continue" the shoppingcart page', () => {

//      // cy.xpath("//label[normalize-space()='60x40 cm']")
//      //     .click()
//      //     cy.wait(300)
      cy.wait(2000)
      cy.xpath("//span[contains(text(),'Alle akzeptieren')]")
      .click()
      cy.wait(2000)
//
      cy.get("#UploadFileDesktop")
      .click()
      cy.wait(1000)
  })


   it('The user on shoppingcart page goes to logincheckout page', () => {
      cy.wait(7000)
      cy.xpath("//button[@id='shoppingcartForm:proceedToCheckout']")
      .click({force: true})
//      .debug()
  })


  it('The user on logincheckout page goes to the checkout page', () => {

  cy.wait(3000)
  cy.xpath("//a[@class='action-register button_action dBlock']")
  .click({force: true})

  })

})

**Reply by jjhelguero** [reply]
If this is all your test code, then you may benefit from reading a few of these pages to make your tests stronger and more resilient.
https://docs.cypress.io/guides/overview/key-differences#Flake-resistant
https://docs.cypress.io/guides/references/best-practices#Having-tests-rely-on-the-state-of-previous-tests

**Comment by vadim3000** [comment]
Thank you, it worked, now just one it()

I figure out, how to split the  code, in End to End purchase scenario in more it()
